adam/language_specific/english/english_language_generator.py

```diff
@@ -331,6 +390,54 @@ def _preposition_for_region_as_goal(self, region: Region[SituationObject]) -> st
                     f"Don't know how to translate {region} to a preposition yet"
                 )
 
+        def _translate_relation_to_verb(
+            self, relation: Relation[SituationObject]
+        ) -> DependencyTreeToken:
+            lexicon_entry = self._unique_lexicon_entry(relation.relation_type)
+            if any(
+                property_ in relation.first_slot.properties
+                for property_ in [IS_SPEAKER, IS_ADDRESSEE]
+            ):
+                word_form = lexicon_entry.base_form
+            elif lexicon_entry.verb_form_3SG_PRS:
+                word_form = lexicon_entry.verb_form_3SG_PRS
+            else:
+                raise RuntimeError(
+                    f"Verb has no 3SG present tense form: {lexicon_entry.base_form}"
+                )
+            verb_dependency_node = DependencyTreeToken(
+                word_form, lexicon_entry.part_of_speech
+            )
+            self.dependency_graph.add_node(verb_dependency_node)
+
+            first_slot_dependency_node = self.objects_to_dependency_nodes[
+                relation.first_slot
+            ]
+            self.dependency_graph.add_edge(
+                first_slot_dependency_node,
+                verb_dependency_node,
+                role=self._translate_argument_role(AGENT),
```

Comment:
I would just specify the roles to be subject and direct object directly; no need to do the translation step.